### PR TITLE
More porting from rep

### PR DESCRIPTION
**Pull request by jefferai**

### CODE REVIEW

File: builtin/credential/app-id/backend.go [hunk 1]
```diff
@@ -17,20 +17,10 @@ func Factory(conf *logical.BackendConfig) (logical.Backend, error) {
...
-	if err != nil {
-		return nil, err
-	}
-
 	var b backend
-	b.Salt = salt
 	b.MapAppId = &framework.PolicyMap{
 		PathMap: framework.PathMap{
 			Name: "app-id",
-			Salt: salt,
```
Comment (briankassouf) [inline builtin/credential/app-id/backend.go]: I could be wrong, but it looks like we lost this salt?

File: builtin/credential/app-id/backend.go [hunk 2]
```diff
@@ -48,7 +38,6 @@ func Backend(conf *logical.BackendConfig) (*framework.Backend, error) {
 
 	b.MapUserId = &framework.PathMap{
 		Name: "user-id",
-		Salt: salt,
```
Comment (briankassouf) [inline builtin/credential/app-id/backend.go]: Same

File: builtin/logical/cassandra/backend.go [hunk 3]
```diff
@@ -107,6 +109,20 @@ func (b *backend) ResetDB(newSession *gocql.Session) {
...
+	switch key {
+	case "config/connection":
+		b.lock.Lock()
+		defer b.lock.Unlock()
+
+		if b.session != nil {
+			b.session.Close()
+		}
+
+		b.session = nil
```
Comment (briankassouf) [inline builtin/logical/cassandra/backend.go]: Could replace this whole branch with `b.ResetDB(nil)`

File: command/mount.go [hunk 4]
```diff
@@ -54,6 +56,7 @@ func (c *MountCommand) Run(args []string) int {
 			DefaultLeaseTTL: defaultLeaseTTL,
 			MaxLeaseTTL:     maxLeaseTTL,
 		},
+		Local: local,
```
Comment (briankassouf) [inline command/mount.go]: In OSS shouldn't this always be `true`?

File: builtin/logical/cassandra/backend.go (hunk 3 above)
Comment (jefferai) [inline builtin/logical/cassandra/backend.go]: True that.

File: command/mount.go (hunk 4 above)
Comment (jefferai) [inline command/mount.go]: I think you mean "always false" :-) But no -- if they switch later the behavior will still be what was set at mount time.

Comment (briankassouf) [inline command/mount.go]: :+1:

**Pull request merged**